### PR TITLE
Feature Engineering

### PR DESCRIPTION
**Pull request by Egjfour**
Added logic for calculating
https://trioml-dsti-project.atlassian.net/jira/core/projects/DGRP/board?selectedIssue=DGRP-8

See notebook for cutoff values and methodology (goodreads-predictor/feature_engineering_explore.ipynb)

model_data_creation pipeline updated to incorporate these features as well

### CODE REVIEW

File: goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/nodes.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
+"""
+This is a boilerplate pipeline 'feature_engineering'
+generated using Kedro 0.19.6
+"""
+from typing import Tuple, List, Union, Dict, Any
```
Comment (Ilyes-Azni) [inline goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/nodes.py:5]: Might change typing to typing_extensions ! As typing is built in for python 3.11 and plus.

File: goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/nodes.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        .first()
+
+        # Map the quantile rank to a string ("Unknown", "Not Well-Known", "Well-Known", "Very Well-Known", "Famous")
+        .assign(AuthorFameLevel = lambda x: x["QuantileRank"].map({1: "1 - Very Low Exposure", 2: "2 - Not Well-Known", 3: "3 - Well-Known",
+                                                                   4: "4 - Popular", 5:"5 - Semi-Famous", 6: "6 - Famous"}))
+        .sort_values("TotalReviews", ascending=False)
+        [['AuthorFameLevel']]
+    )
+
+    return author_fame_levels
```
Comment (Ilyes-Azni) [inline goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/nodes.py:94]: Nice one! Having this is awesome as it is clearly a must to a priori define what is a good book

**Review by Ilyes-Azni — APPROVED** [review]
Awesome. All the feature engineering is logic and simple (considering the math and the reflexion) I really do think that the more intuitive it gets the more valuable those features might be. Nice work Eddie 😀

**Review by ismail-collab — APPROVED** [review]

File: goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/nodes.py (hunk 1 above)
Comment (Egjfour) [reply goodreads-predictor/src/goodreads_predictor/pipelines/feature_engineering/nodes.py]: Done, thanks for the feedback!

**Pull request merged**